instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -363,6 +371,19 @@ def _instrumented_lambda_handler_call(
                 "TracerProvider was missing `force_flush` method. This is necessary in case of a Lambda freeze and would exist in the OTel SDK implementation."
             )
 
+        rem = flush_timeout - (time.time()-now)*1000
+        if rem > 0:
+            _meter_provider = meter_provider or get_meter_provider()
+            try:
+                # NOTE: `force_flush` before function quit in case of Lambda freeze.
+                # Assumes we are using the OpenTelemetry SDK implementation of the
+                # `MeterProvider`.
+                _meter_provider.force_flush(rem)
+            except Exception:  # pylint: disable=broad-except
+                logger.error(
+                    "MeterProvider was missing `force_flush` method. This is necessary in case of a Lambda freeze and would exist in the OTel SDK implementation."
+                )
```

Comment:
This message will be incorrect because `force_flush` raises an exception when it timed out or the configured metrics reader collection fails. If you want to check for the existence of the method use `hasattr(meter_provider, 'force_flush')`.

Reply:
I was mirroring the `TracerProvider` messaging above.  Should I change both?  What is the correct pattern, to check `hasattr()` and then to `try` to `force_flush()`?

Reply:
I would use `hasattr` and then `force_flush` as you suggested 👍🏻